### PR TITLE
Switch Snippets to use snippet_variables

### PR DESCRIPTION
**Pull request by mheap**
## Description

Replace all code snippet variable tables with the new `snippet_variables` markdown helper

## Deploy Notes

N/A

### CODE REVIEW

File: _documentation/en/account/code-snippets/create-a-secret.md [hunk 1]
```diff
@@ -9,16 +9,17 @@ To create a new API secret, you must send a `POST` request to the secret managem
...
+-   Minimum 1 upper case character
+-   Minimum 1 digit
 
-Key | Description
- -- | --
-`VONAGE_API_KEY` | The API key of the account.
-`NEW_SECRET` | The new API secret for the API key.
+```snippet_variables
+- VONAGE_API_KEY
+- VONAGE_API_SECRET
```
Comment (bencgreenberg) [inline _documentation/en/account/code-snippets/create-a-secret.md:20]: The api secret currently is not part of this snippet, and the snippets use a hard coded value there.

File: _documentation/en/account/code-snippets/fetch-a-secret.md [hunk 2]
```diff
@@ -8,10 +8,12 @@ navigation_weight: 4
 This will return the details of a single secret. The value of the secret will
 never be shown.
 
-Key | Description
- -- | --
-`VONAGE_API_KEY` | The API key of the account to fetch secrets for. Can be the same as authenticating account or it can be a subaccount.
-`VONAGE_SECRET_ID` | The ID of the secret to retrieve.
+```snippet_variables
+- VONAGE_API_KEY
+- VONAGE_API_SECRET
```
Comment (bencgreenberg) [inline _documentation/en/account/code-snippets/fetch-a-secret.md:13]: The same is true as above. The API secret and account ID are not currently set dynamically here.

File: _documentation/en/dispatch/code-snippets/before-you-begin.md [hunk 3]
```diff
@@ -22,23 +22,23 @@ source: _partials/reusable/prereqs.md
...
 All phone numbers are in E.164 format.
 
-Key | Description
--- | --
-`VONAGE_NUMBER` | Replace with your Vonage Number. E.g. 447700900000
-`FROM_NUMBER` | Replace with number you are sending from. E.g. 447700900002
-`TO_NUMBER` | Replace with the number you are sending to. E.g. 447700900001
+```snippet_variables
+- VONAGE_NUMBER
+- TO_NUMBER.DISPATCH
```
Comment (bencgreenberg) [inline _documentation/en/dispatch/code-snippets/before-you-begin.md:39]: Is there a reason why we reversed the order here? Does it make a difference?

**Review by bencgreenberg — APPROVED** [review]
I went through a whole lot of these and all look good. I left a few comments/questions on a couple of them.

File: _documentation/en/account/code-snippets/list-all-secrets.md [hunk 4]
```diff
@@ -8,6 +8,12 @@ navigation_weight: 3
 This will return all secrets, along with their `id` and `created_at` time. The
 value of the secret will never be shown
 
+```snippet_variables
+- VONAGE_API_KEY
+- VONAGE_API_SECRET
+- ACCOUNT_ID
```
Comment (fabianrbz) [inline _documentation/en/account/code-snippets/list-all-secrets.md:14]: I don't think `ACCOUNT_ID` is used in this snippet.

File: _documentation/en/account/code-snippets/revoke-a-secret.md [hunk 5]
```diff
@@ -9,10 +9,12 @@ To create a new API secret, you must send a `DELETE` request to the secret manag
...
 You must have at least one API secret at all times.
 
-Key | Description
- -- | --
-`VONAGE_API_KEY` | The API key of the account.
-`VONAGE_SECRET_ID` | The ID of the secret to delete.
+```snippet_variables
+- VONAGE_API_KEY
+- VONAGE_API_SECRET
+- ACCOUNT_ID
```
Comment (fabianrbz) [inline _documentation/en/account/code-snippets/revoke-a-secret.md:15]: same here

File: _documentation/en/application/code-snippets/delete-application.md [hunk 6]
```diff
@@ -12,11 +12,11 @@ In this code snippet you will see how to delete the specified Application.
...
 
-Key | Description
--- | --
-`VONAGE_API_KEY` | Your Vonage API key.
-`VONAGE_API_SECRET` | Your Vonage API secret.
-`VONAGE_APPLICATION_ID` | The application ID for the Vonage application you want to delete.
+```snippet_variables
+- VONAGE_API_KEY
+- VONAGE_API_SECRET
+- VONAGE_APPLICATION_ID
```
Comment (fabianrbz) [inline _documentation/en/application/code-snippets/delete-application.md:18]: we lost `for the Vonage application you want to delete.` from `VONAGE_APPLICATION_ID` I don't think that's much of an issue, but something we might want to consider.

File: _documentation/en/application/code-snippets/get-application.md [hunk 7]
```diff
@@ -12,11 +12,11 @@ In this code snippet you will see how to retrieve details of the specified Appli
...
 
-Key | Description
--- | --
-`VONAGE_API_KEY` | Your Vonage API key.
-`VONAGE_API_SECRET` | Your Vonage API secret.
-`VONAGE_APPLICATION_ID` | The application ID for the Vonage application you want to get.
+```snippet_variables
+- VONAGE_API_KEY
+- VONAGE_API_SECRET
+- VONAGE_APPLICATION_ID
```
Comment (fabianrbz) [inline _documentation/en/application/code-snippets/get-application.md:18]: same here `VONAGE_APPLICATION_ID`

File: _documentation/en/application/code-snippets/update-application.md [hunk 8]
```diff
@@ -12,12 +12,12 @@ In this code snippet you will see how to update an Application.
...
--- | --
-`VONAGE_API_KEY` | Your Vonage API key.
-`VONAGE_API_SECRET` | Your Vonage API secret.
-`VONAGE_APPLICATION_ID` | The application ID for the Vonage application you want to update.
-`NAME` | The new name for the Vonage application you want to update.
+```snippet_variables
+- VONAGE_API_KEY
+- VONAGE_API_SECRET
+- VONAGE_APPLICATION_ID
+- NAME.UPDATE.APPLICATION
```
Comment (fabianrbz) [inline _documentation/en/application/code-snippets/update-application.md:19]: Note: We need to update all the snippets, given that they are using either the `VONAGE_APPLICATION_ID` or a hardcoded name.

File: config/code_snippet_variables.yml [hunk 9]
```diff
@@ -1,6 +1,142 @@
...
+# Audit API
+EVENT_UUID: The UUID of the audit event you want to retrieve.
+SEARCH_TEXT: Some JSON compatible text to search for. For example, "number".
+DATE_FROM: Audit events starting from this ISO-8601 datetime value. For example, "2018-07-01".
+DATE_TO: Audit events up to this ISO-8601 datetime value. For example, "2018-08-01".
+
+# Conversation
+CONV_NAME: The unique name of the Conversation.
+CONV_DISPLAY_NAME: The display name of the Conversation.
+CONVERSATION_ID: The ID of the Conversation
```
Comment (fabianrbz) [inline config/code_snippet_variables.yml:41]:
```suggestion
CONVERSATION_ID: The ID of the Conversation.
```

File: config/code_snippet_variables.yml [hunk 10]
```diff
@@ -1,6 +1,142 @@
...
+CONVERSATION_ID.GET: The ID of the Conversation to be retrieved.
+CONVERSATION_ID.DELETE: The ID of the Conversation to be deleted.
+CONV_NEW_NAME: The new name.
+CONV_NEW_DISPLAY_NAME: The new display name
```
Comment (fabianrbz) [inline config/code_snippet_variables.yml:45]:
```suggestion
CONV_NEW_DISPLAY_NAME: The new display name.
```

File: config/code_snippet_variables.yml [hunk 11]
```diff
@@ -1,6 +1,142 @@
...
+MEMBER_ID: The unique ID of the Member
+EVENT_ID: The unique ID of the Event
+LEG_ID: The unique ID of the Leg
```
Comment (fabianrbz) [inline config/code_snippet_variables.yml:48]:
```suggestion
LEG_ID: The unique ID of the Leg.
```

File: config/code_snippet_variables.yml [hunk 12]
```diff
@@ -1,6 +1,142 @@
...
+USER_ID: The unique ID of the User
```
Comment (fabianrbz) [inline config/code_snippet_variables.yml:49]:
```suggestion
USER_ID: The unique ID of the User.
```

File: config/code_snippet_variables.yml [hunk 13]
```diff
@@ -1,6 +1,142 @@
...
+
+# Conversation
+CONV_NAME: The unique name of the Conversation.
+CONV_DISPLAY_NAME: The display name of the Conversation.
+CONVERSATION_ID: The ID of the Conversation
+CONVERSATION_ID.GET: The ID of the Conversation to be retrieved.
+CONVERSATION_ID.DELETE: The ID of the Conversation to be deleted.
+CONV_NEW_NAME: The new name.
+CONV_NEW_DISPLAY_NAME: The new display name
+MEMBER_ID: The unique ID of the Member
```
Comment (fabianrbz) [inline config/code_snippet_variables.yml:46]:
```suggestion
MEMBER_ID: The unique ID of the Member.
```

File: config/code_snippet_variables.yml [hunk 14]
```diff
@@ -1,6 +1,142 @@
...
+EVENT_ID: The unique ID of the Event
+LEG_ID: The unique ID of the Leg
+USER_ID: The unique ID of the User
+USER_NAME: The unique name of the User
```
Comment (fabianrbz) [inline config/code_snippet_variables.yml:50]:
```suggestion
USER_NAME: The unique name of the User.
```

File: config/code_snippet_variables.yml [hunk 15]
```diff
@@ -1,6 +1,142 @@
...
+USER_DISPLAY_NAME: The display name of the User.
+USER_NEW_NAME: The new name of the User
```
Comment (fabianrbz) [inline config/code_snippet_variables.yml:52]:
```suggestion
USER_NEW_NAME: The new name of the User.
```

File: config/code_snippet_variables.yml [hunk 16]
```diff
@@ -1,6 +1,142 @@
...
-NEXMO_API_SECRET:
-  description: You can find this in your account overview
-TO_NUMBER:
-  description: The number you are sending the SMS to in E.164 format. For example `447700900000`.
\ No newline at end of file
+# Contribution guide examples
+NEXMO_API_KEY: Your Nexmo API key (see it on [your dashboard](http://dashboard.nexmo.com/))
+
+# Generic
+VONAGE_API_KEY: Your Vonage API key (see it on [your dashboard](http://dashboard.nexmo.com/))
```
Comment (fabianrbz) [inline config/code_snippet_variables.yml:5]:
```suggestion
VONAGE_API_KEY: Your Vonage API key (see it on [your dashboard](http://dashboard.nexmo.com/)).
```

File: config/code_snippet_variables.yml [hunk 17]
```diff
@@ -1,6 +1,142 @@
...
+VONAGE_API_SECRET: Your Vonage API secret (also available on [your dashboard](http://dashboard.nexmo.com/))
```
Comment (fabianrbz) [inline config/code_snippet_variables.yml:6]:
```suggestion
VONAGE_API_SECRET: Your Vonage API secret (also available on [your dashboard](http://dashboard.nexmo.com/)).
```

File: _documentation/en/dispatch/code-snippets/send-an-mms-with-failover.md [hunk 18]
```diff
@@ -18,11 +18,11 @@ In the Workflow object, message objects can be placed in any order to suit your
...
 Ensure the following variables are set to your required values using any convenient method:
 
-Key | Description
--- | --
-`VONAGE_APPLICATION_ID` | The ID of the application that you created.
-`FROM_NUMBER` | The phone number you are sending the MMS from (US Short Code only).
-`TO_NUMBER` | The phone number you are sending the MMS to.
+```snippet_variables
+- VONAGE_APPLICATION_ID
+- FROM_NUMBER.DISPATCH
```
Comment (fabianrbz) [inline _documentation/en/dispatch/code-snippets/send-an-mms-with-failover.md:23]: we were using the text that  is defined in `FROM_NUMBER.MMS.MESSAGES`  for `FROM_NUMBER.DISPATCH`

File: _documentation/en/messages/code-snippets/viber/send-image.md [hunk 19]
```diff
@@ -16,14 +16,14 @@ For a step-by-step guide to this topic, you can read our tutorial [Sending Viber
...
-`MESSAGES_API_URL` | For production use the Messages API endpoint is `https://api.nexmo.com/v0.1/messages`. For sandbox testing the Messages API endpoint is `https://messages-sandbox.nexmo.com/v0.1/messages`.
-`VONAGE_APPLICATION_ID` | The ID of the application that you created.
-`VIBER_SERVICE_MESSAGE_ID` | Your Viber Service Message ID. For sandbox testing this is `16273`.
-`TO_NUMBER` | The phone number you are sending the message to.
-`IMAGE_URL` | The URL of the image you want to send.
+```snippet_variables
+- BASE_URL.MESSAGES
+- MESSAGES_API_URL
+- VONAGE_APPLICATION_ID
+- VIBER_SERVICE_MESSAGE_ID
```
Comment (fabianrbz) [inline _documentation/en/messages/code-snippets/viber/send-image.md:23]: Note: `VIBER_SERVICE_MESSAGE_ID` was:

`Your Viber Service Message ID. For sandbox testing this is 16273.`

File: _documentation/en/messages/code-snippets/whatsapp/send-audio.md [hunk 20]
```diff
@@ -13,13 +13,13 @@ In this code snippet you learn how to send a WhatsApp audio message using the Me
...
--- | --
-`BASE_URL` | For production use the base URL is `https://api.nexmo.com/`. For sandbox testing the base URL is `https://messages-sandbox.nexmo.com/`.
-`MESSAGES_API_URL` | For production use the Messages API endpoint is `https://api.nexmo.com/v0.1/messages`. For sandbox testing the Messages API endpoint is `https://messages-sandbox.nexmo.com/v0.1/messages`.
-`WHATSAPP_NUMBER` | The WhatsApp number that has been allocated to you by Vonage. For sandbox testing the number is `14157386170`.
-`TO_NUMBER` | The phone number you are sending the message to.
-`AUDIO_URL` | The URL of the audio file to send.
+```snippet_variables
+- BASE_URL.MESSAGES
+- MESSAGES_API_URL
+- WHATSAPP_NUMBER
```
Comment (fabianrbz) [inline _documentation/en/messages/code-snippets/whatsapp/send-audio.md:19]: Note:
`WHATSAPP_NUMBER` was:

`The WhatsApp number that has been allocated to you by Vonage. For sandbox testing the number is 14157386170.`

File: config/code_snippet_variables.yml [hunk 21]
```diff
@@ -1,6 +1,142 @@
...
+AMOUNT.VERIFY: How much the payment is for (always in Euro)
+WORKFLOW_ID.VERIFY: Choose a workflow (number between 1 and 7), these are defined in the [workflows guide](/verify/guides/workflows-and-events)
+BRAND_NAME.VERIFY: Included in the message to explain who is confirming the phone number
+CODE.VERIFY: The code the user supplies as having been sent to them
+
+# Voice
+TO_NUMBER.VOICE: The number you are calling. E.g. `447700900001`
+SECOND_NUMBER.VOICE: The number you are forwarding to. E.g. `447700900002`
+YOUR_SECOND_NUMBER.VOICE: The number you are connecting to. E.g. `447700900002`
+UUID.VOICE: The UUID of the call leg
```
Comment (fabianrbz) [inline config/code_snippet_variables.yml:135]: could we make `leg` a link to the concept? i.e. `/conversation/concepts/leg#leg`

File: _documentation/en/account/code-snippets/create-a-secret.md (hunk 1 above)
Comment (mheap) [reply _documentation/en/account/code-snippets/create-a-secret.md:20]: `VONAGE_API_SECRET` is used in the `Initialize your dependencies` section for auth. Some (not all) snippets use a hardcoded value but should instead use `NEW_SECRET`. The snippets should be updated.

File: _documentation/en/account/code-snippets/fetch-a-secret.md (hunk 2 above)
Comment (mheap) [reply _documentation/en/account/code-snippets/fetch-a-secret.md:13]: These are used for authentication

File: _documentation/en/account/code-snippets/list-all-secrets.md (hunk 4 above)
Comment (mheap) [reply _documentation/en/account/code-snippets/list-all-secrets.md:14]: It is used in the `curl` sample and should be used in all to differentiate between `VONAGE_API_KEY` for authentication and `ACCOUNT_ID` to target the account to list secrets for. In most cases they will be the same

File: _documentation/en/account/code-snippets/revoke-a-secret.md (hunk 5 above)
Comment (mheap) [reply _documentation/en/account/code-snippets/revoke-a-secret.md:15]: It is used in the `curl` sample and should be used in all to differentiate between `VONAGE_API_KEY` for authentication and `ACCOUNT_ID` to target the account to list secrets for. In most cases they will be the same

File: _documentation/en/application/code-snippets/delete-application.md (hunk 6 above)
Comment (mheap) [reply _documentation/en/application/code-snippets/delete-application.md:18]: I'm not sure how much it adds. If we add it we need to define `VONAGE_APPLICATION_ID.APPLICATION.DELETE`, `VONAGE_APPLICATION_ID.APPLICATION.UPDATE` etc and I'm not sure there's enough value there

File: _documentation/en/application/code-snippets/get-application.md (hunk 7 above)
Comment (mheap) [reply _documentation/en/application/code-snippets/get-application.md:18]: As above, so resolving to focus on a single example and I'll update them all if we decide to

File: _documentation/en/dispatch/code-snippets/before-you-begin.md (hunk 3 above)
Comment (mheap) [reply _documentation/en/dispatch/code-snippets/before-you-begin.md:39]: The `TO_NUMBER` ends in `1` and the `FROM_NUMBER` ends in `2`. No other reason

File: _documentation/en/dispatch/code-snippets/send-an-mms-with-failover.md (hunk 18 above)
Comment (mheap) [reply _documentation/en/dispatch/code-snippets/send-an-mms-with-failover.md:23]: Great catch! Updated in the next commit

File: _documentation/en/messages/code-snippets/viber/send-image.md (hunk 19 above)
Comment (mheap) [reply _documentation/en/messages/code-snippets/viber/send-image.md:23]: Resolved in the next commit

File: _documentation/en/messages/code-snippets/whatsapp/send-audio.md (hunk 20 above)
Comment (mheap) [reply _documentation/en/messages/code-snippets/whatsapp/send-audio.md:19]: Resolved in the next commit

**Pull request merged**